libnetwork/portmapper/proxy_linux.go

```diff
@@ -61,6 +61,10 @@ func (p *proxyCommand) Start() error {
 	}
 	w.Close()
 
+	go func() {
+		p.cmd.Wait()
```

Comment:
Given `Stop()` also calls `p.cmd.Wait()` doesn't this introduce a potential race such that `Stop()` might return `exec: Wait was already called`?

Reply:
```	
if p.cmd.Process != nil {
		if err := p.cmd.Process.Signal(os.Interrupt); err != nil {
			return err
		}
		return p.cmd.Wait()
	}
```
if   p.cmd.Process.Signal(os.Interrupt)   err is "process already finished" will not go to  p.cmd.Wait()

Reply:
Here's the sequence I'm worried about:

1. `Stop()` invokes `p.cmd.Process.Signal(os.Interrupt)` and the process exits
2. this new goroutine picks up the `Wait()` and reaps the process
3. `Stop()` then invokes `p.cmd.Wait()` and returns an error instead of a successful exit code

Reply:
change from 
type proxyCommand struct {
	cmd *exec.Cmd
}
to 
type proxyCommand struct {
	mu         sync.Mutex
	cmd *exec.Cmd
}
and add lock() will ok?

Reply:
done